build-config/Makefile

```diff
@@ -162,7 +162,7 @@ ifneq (,$(MAKECMDGOALS))
       $(error Unable to find valid machine configuration directory.)
     endif
     include $(MACHINEDIR)/machine.make
-    ARCHDIR ?= $(realpath ./arch)
+    ARCHDIR ?= $(realpath ./arch/$(ONIE_ARCH))
```

Comment:
I was able to get it working with the following change:
```
ARCHDIR ?= $(shell realpath ./arch/$(ONIE_ARCH))
```

Except now, I get the following error:
```
Makefile:166: [snip]/x86_64.make: No such file or directory
make: *** No rule to make target '[snip]/x86_64.make.  Stop.
```

This is because the arch directory is named ```x86_x64``` vs ```x86_64```.